presto-bytecode/src/main/java/com/facebook/presto/bytecode/MethodDefinition.java

```diff
@@ -70,6 +74,15 @@ public MethodDefinition(
             Iterable<Parameter> parameters
     )
     {
+        // Code generation could failed if parameter count is 255 or more. Given that fixing this directly might be difficult, handle it by capping the maximum number of arguments
+        // to 250.
+        int parameterCount = Iterables.size(parameters);
+        checkArgument(
+                parameterCount > MAX_ALLOWED_VARARG_ARGUMENTS,
+                "Number of arguments (%d) for vararg functions exceeds limit (%d)",
```

Comment:
This doesn't have anything to do with varargs. The message should be something like: "Methods may have at most 255 parameters"

(use the term "parameters" to be consistent with the JVMS)